packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandExecution,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as AdmZip from 'adm-zip';
+import { SpawnOptions } from 'child_process';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as shell from 'shelljs';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath } from '../util';
+import {
+  CommonDirDirectoryDiffer,
+  DirectoryDiffer,
+  DirectoryDiffResults
+} from './directoryDiffer';
+
+export interface ConflictDetectionConfig {
+  usernameOrAlias: string;
+  packageDir: string;
+  manifest: string;
+}
+
+export class ConflictDetector {
+  private differ: DirectoryDiffer;
+  private static instance: ConflictDetector;
+
+  constructor(differ?: DirectoryDiffer) {
+    this.differ = differ || new CommonDirDirectoryDiffer();
+  }
+
+  public static getInstance(): ConflictDetector {
+    if (!ConflictDetector.instance) {
+      ConflictDetector.instance = new ConflictDetector(
+        new CommonDirDirectoryDiffer()
+      );
+    }
+    return ConflictDetector.instance;
+  }
+
+  private readonly relativeMetdataTempPath = path.join(
```

Comment:
Do we want to do like what we did with the diff plugin and utilize `os.temp` so that it's periodically cleared?

Reply:
The directory is removed when the conflict call is finished.

Reply:
I know we're clearing out the temporary directory we're creating, but I think utilizing the os.temp directory might be safer.

Reply:
I'll look into using os.temp.

Reply:
Done